test/swar/BasicOperations.cpp

```diff
@@ -33,6 +31,71 @@ using S32_32 = SWAR<32, uint32_t>;
 
 using S64_64 = SWAR<64, uint64_t>;
 
+static_assert(SWAR<16, u64>::MaxUnsignedLaneValue == 65535);
+static_assert(SWAR<16, u32>::MaxUnsignedLaneValue == 65535);
+static_assert(SWAR<8, u32>::MaxUnsignedLaneValue == 255);
+static_assert(SWAR<4, u32>::MaxUnsignedLaneValue == 15);
+static_assert(SWAR<2, u32>::MaxUnsignedLaneValue == 3);
+
+static_assert(SWAR{Literals<32, u64>, {2, 1}}.value() == 0x00000002'00000001);
+static_assert(SWAR{Literals<32, u64>, {1, 2}}.value() == 0x00000001'00000002);
+
+static_assert(SWAR{Literals<16, u64>, {4, 3, 2, 1}}.value() == 0x0004'0003'0002'0001);
+static_assert(SWAR{Literals<16, u64>, {1, 2, 3, 4}}.value() == 0x0001'0002'0003'0004);
+
+static_assert(SWAR{Literals<16, u32>, {2, 1}}.value() == 0x0002'0001);
+static_assert(SWAR{Literals<16, u32>, {1, 2}}.value() == 0x0001'0002);
+
+static_assert(SWAR{Literals<8, u32>, {4, 3, 2, 1}}.value() == 0x04'03'02'01);
+static_assert(SWAR{Literals<8, u32>, {1, 2, 3, 4}}.value() == 0x01'02'03'04);
+static_assert(SWAR{Literals<8, u32>, {1, 2, 3, 4}}.value() == 0x01'02'03'04);
+
+static_assert(SWAR{Literals<8, u16>, {2, 1}}.value() == 0x0201);
+static_assert(SWAR{Literals<8, u16>, {1, 2}}.value() == 0x0102);
+
+static_assert(SWAR{Literals<4, u8>, {2, 1}}.value() == 0x21);
+static_assert(SWAR{Literals<4, u8>, {1, 2}}.value() == 0x12);
+
+// Little-endian
+static_assert(SWAR{Literals<16, u64>, {1, 2, 3, 4}}.at(0) == 4);
+static_assert(SWAR{Literals<16, u64>, {1, 2, 3, 4}}.at(1) == 3);
+
+// Macro required because initializer lists are not constexpr
+#define ARRAY_TEST(SwarType, ...)                                              \
+    static_assert([]() {                                                       \
+        using S = SwarType;                                                    \
+        constexpr auto arry = std::array{__VA_ARGS__};                         \
+        constexpr auto test_array = S{S::Literal, {__VA_ARGS__}}.to_array();   \
+        static_assert(arry.size() == S::Lanes);                                \
+        for (auto i = 0; i < S::Lanes; ++i) {                                  \
```

Comment:
does operator=() not work correctly for std::array?

Reply:
it would seem not for constexpr things in cpp17